__mocks__/@google-cloud/storage.js

```diff
@@ -0,0 +1,85 @@
+// This was mostly stolen from https://gist.github.com/nfarina/90ba99a5187113900c86289e67586aaa. Thanks!
+// @flow
+
+//
+// Quick & Dirty Google Cloud Storage emulator for tests. Requires
+// `stream-buffers` from npm. Use it like this:
+//
+// `new MockStorage().bucket('my-bucket').file('my_file').createWriteStream()`
+//
+
+import { Readable } from 'stream';
+import { Concatenator } from '../../src/utils/streams';
+
+class MockStorage {
+  static buckets: { [name: string]: MockBucket } = {};
+  static cleanUp() {
+    this.buckets = {};
+  }
+
+  bucket(name: string) {
+    return (
+      MockStorage.buckets[name] ||
+      (MockStorage.buckets[name] = new MockBucket(name))
+    );
+  }
+}
+
+class MockBucket {
+  name: string;
+  files: { [path: string]: MockFile };
+
+  constructor(name: string) {
+    this.name = name;
+    this.files = {};
+  }
+
+  file(path: string) {
+    return this.files[path] || (this.files[path] = new MockFile(path));
+  }
+}
+
+type Metadata = {
+  +metadata: { ... },
+  ...
+};
```

Comment:
In case you wonder what these `...` are, this is to represent explicitely inexact objects in recent versions of flow.